File: VB/E5023/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports E5023.Models
Imports E5023.Reports

Namespace E5023.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function DocumentViewerPartial() As ActionResult
			Dim model = Category.GetCategories()
			Return PartialView("_DocumentViewerPartial", model)
		End Function

		Public Function DocumentViewerPartialExport() As ActionResult
			Dim report As New CategoriesReport()
			report.DataSource = Category.GetCategories()
			Return ReportViewerExtension.ExportTo(report)
		End Function
	End Class
End Namespace

File: CS/E5023/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using E5023.Models;
using E5023.Reports;

namespace E5023.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }        

        public ActionResult DocumentViewerPartial() {
            var model = Category.GetCategories();
            return PartialView("_DocumentViewerPartial", model);
        }

        public ActionResult DocumentViewerPartialExport() {
            CategoriesReport report = new CategoriesReport();
            report.DataSource = Category.GetCategories();
            return ReportViewerExtension.ExportTo(report);
        }
    }
}